xapian-core/api/omenquire.cc

```diff
@@ -452,6 +464,23 @@ MSet::Internal::read_docs() const
     requested_docs.clear();
 }
 
+void
+MSet::Internal::set_new_weights(double wt, int i) {
```

Comment:
You should be able to set it directly from within any `MSet::Internal` method. However, if you really cannot, you should overload `MSet::Internal::set_item_weight()` rather than call one `set_item_weight_()`.